Feature/search type ahead

sheldonkwoodward opened the pull request.

sheldonkwoodward committed:
Typeahead for search results working.

sheldonkwoodward committed:
Moved typeahead result updater into fat arrow function.

sheldonkwoodward committed:
Merge branch 'develop' into feature/search-type-ahead

sheldonkwoodward committed:
Added search button and typeahead for search.

sheldonkwoodward committed:
Typeahead now supports majors, not minors. Search button styling.

sheldonkwoodward committed:
Added formatting for majors and minors. Search button added, looks like search field.

sheldonkwoodward committed:
Added search button icon. Fixed slight space between fill and border os search area.

sheldonkwoodward committed:
Merge branch 'develop' into feature/search-type-ahead

sheldonkwoodward committed:
Final cleanup.

sheldonkwoodward committed:
Final cleanup 2.

ryanrabello commented on src/app/routes/search/search.component.ts:
```diff
@@ -2,36 +2,86 @@ import { Subscription } from 'rxjs';
+    // get all profile view info
+    var query = this.typedQuery || "";
+    this.typeaheadSub = this.rs.getWithSub('/search/'+ CURRENT_YEAR + "/" + query , (data) => {
+      this.allProfiles = data.results.sort((p1,p2) => {
```
Server now responds with data already sorted. 🙌 So we shouldn't need this. (I'll look into it).

ryanrabello committed:
Sorting is now done one the server
I also added a comment.

ryanrabello committed:
SelectItem event triggers runsearch()
This means that clicking on a name in the typeahead area will search for that name.

ryanrabello reviewed (approved):
Looks good.
